executor/analyze.go

```diff
@@ -135,6 +161,34 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	if err != nil {
 		return err
 	}
+	if needGlobalStats {
+		for _, info := range globalStatsMap {
+			globalStats, succ := statsHandle.MergePartitionStats2GlobalStats(infoschema.GetInfoSchema(e.ctx), info.tableID, info.isIndex, info.idxID)
+			if succ {
+				for i := 0; i < globalStats.Num; i++ {
+					hg := globalStats.Hg[i]
+					cms := globalStats.Cms[i]
+					topN := globalStats.TopN[i]
+					if hg == nil || cms == nil || topN == nil {
+						succ = false
+						break
+					}
+					err = statsHandle.SaveStatsToStorage(info.tableID, globalStats.Count, info.isIndex, hg, cms, topN, info.statsVersion, 1)
+					if err != nil {
+						logutil.Logger(ctx).Error("save global-level stats to storage failed", zap.Error(err))
+						succ = false
+					}
+				}
+			}
+			if !succ {
```

Comment:
If build the global-level stats failed, we will report a warning.

Reply:
I prefer to return an error here, since this progress may be triggered by `auto-analyze` and warnings will be ignored then.